ios/brave-ios/Sources/BraveWallet/Crypto/CryptoTabsView.swift

```diff
@@ -135,6 +147,9 @@ struct CryptoTabsView<DismissContent: ToolbarContent>: View {
       tabBarController.tabBar.standardAppearance = appearance
       tabBarController.tabBar.scrollEdgeAppearance = appearance
     })
+    .onChange(of: cryptoStore.viewActivityToggle) { _ in
+      selectedTab = .activity
```

Comment:
yes. that's why i named it as a toggle instead of a boolean type of name. if it's still confusing. i can store an optional tab instead.

Reply:
Did you try moving the `selectedTab` from `CryptoTabsView` into the shared observed object so we don’t have this duplicated property?

Reply:
For example, storing the `TabView` selection as an `@Published` inside the `ObservableObject` instead of `@State` in the view itself so all views can assign it:
```
enum Tab {
  case first, second
}

class Store: ObservableObject {
  @Published var tab: Tab = .first
}

struct TabsView: View {
  @ObservedObject private var store = Store()
  
  var body: some View {
    TabView(selection: $store.tab) {
      FirstTabView(store: store)
        .tabItem {  Text("First") }
        .tag(Tab.first)
      
      SecondTabView(store: store)
        .tabItem { Text("Second") }
        .tag(Tab.second)
    }
  }
}

struct FirstTabView: View {
  var store: Store
  
  var body: some View {
    Color.red
      .onTapGesture { store.tab = .second }
  }
}

struct SecondTabView: View {
  var store: Store
  
  var body: some View {
    Color.blue
      .onTapGesture { store.tab = .first }
  }
}
```
This way there is just one 'selectedTab' property, and any view with access to the `Store: ObservableObject` can change the selected tab. Then you don't need any of the `onChange` modifiers in the view itself to update based on the `ObservedObject`'s published property.

Reply:
ah you meant a different store? i moved it to `CryptoStore` in [a2858cf](https://github.com/brave/brave-core/pull/26146/commits/a2858cf32988de6facbc623c7ed0fcde9656743d)